docs/core/tutorials/unloadability-howto.md

```diff
@@ -0,0 +1,359 @@
+    alcWeakRef = new WeakReference(alc, trackResurrection: true);
+
+    Assembly a = alc.LoadFromAssemblyPath(assemblyPath);
+
+    var args = new object[1] { new string[] {"Hello"}};
+    int result = (int)a.EntryPoint.Invoke(null, args);
+    alc.Unload();
+
+    return result;
+}
+```
+Now we can run this function to Load, execute and Unload the assembly.
+```C#
+WeakReference testAlcWeakRef;
+int result = ExecuteAndUnload("absolute/path/to/your/assembly", out testAlcWeakRef);
+```
+However, the unloading doesn't complete immediately. As I've already mentioned, it relies on GC to collect all the objects from the test assembly (etc.). In many cases, it is not necessary to wait for the unload completion. However there are cases where it is useful to know that the unload has finished. For example, you may want to delete the assembly file that was loaded into the custom context from disk. In such case, the following code snippet can be used. It triggers a GC and waits for pending finalizers in a loop until the weak reference to the custom AssemblyLoadContext is set to null, indicating the target object was collected. Please note that in most cases, just one pass through the loop is required. However for more complex cases where objects created by the code running in the AssemblyLoadContext have finalizers, more passes may be needed.
+```C#
+for (int i = 0; testAlcWeakRef.IsAlive && (i < 10); i++)
+{
+    GC.Collect();
+    GC.WaitForPendingFinalizers();
+}
+```
+### The Unloading event
+In some cases, it may be necessary for the code loaded into a custom AssemblyLoadContext to perform some cleanup when the unloading is initiated. For example, it may need to stop threads, cleanup some strong GC handles, etc. The `Unloading` event can be used in such cases. A handler that performs the necessary cleanup can be hooked to this event.
+### Troubleshooting unloadability issues
+While the previous paragraphs may sound like it is all easy and simple, it is often not the case. Due to the cooperative nature of the unloading, it is easy to forget about references keeping the stuff in the custom AssemblyLoadContext alive and preventing unload. Here is a summary of things (some of them non-obvious) that can hold the references:
+* Regular reference held from outside of the custom AssemblyLoadContext  or a strong / pinning GC handle. Such reference may point to:
+  * An assembly loaded into the custom `AssemblyLoadContext`
+  * Type from such an assembly
+  * Instance of a type  from such an assembly
+* Threads running code from an assembly loaded into the custom AssemblyLoadContext
+* Instances of non-collectible AssemblyLoadContext
+* Pending RegisteredWaitHandle instances
+The above mentioned reference can come from any object with root in a stack slot or processor register (method locals, either explicitly created by the user code or implicitly by the JIT), a static variable or strong / pinning GC handle. TODO: explain what root means
```

Comment:
>TODO: explain what roo means

Is there a nice definition in the BOTR we can reference? If not, I would add one there.